tasks/main.yaml

```diff
@@ -22,7 +22,8 @@
 
 - name: Wait for fio benchmark to complete its execution
   ansible.builtin.service_facts:
-  until: ansible_facts.services is defined and ansible_facts.services['fio.service']['state'] != "running"
+  register: services_state
+  until: services-state.ansible_facts.services is defined and services_state.ansible_facts.services['fio.service'].state != "running"
```

Comment:
Ishan, there is typo here at `services-state`, not matching with register variable :)

Reply:
this is fixed here - https://github.com/mcharanrm/ansible-role-fio-benchmark/pull/3/commits/b93f19bdb236cd5c9440fee12f50f55a8babb95c